### ISSUE TITLE
Augmentation Transform | Patch Shuffling (ETA 8/27)

### ISSUE THREAD

**Issue by finalelement** [Project-MONAI/MONAI#2701]
Local pixel/voxel shuffling is a an augmentation that can be used during training, its essentially a way to add noise to the image. In brief the a small sub-block of pixels/voxels are selected from the entire image and then local pixels are shuffled to generate a noisy image

The expectation is to send an image and the transform should be able to shuffle pixels given the hyper-parameters of how many blocks should be shuffled

It should also be noted that pytorch also offers a Pixel Shuffle ([https://pytorch.org/docs/stable/generated/torch.nn.PixelShuffle.html](https://pytorch.org/docs/stable/generated/torch.nn.PixelShuffle.html)). That should not be confused with this one, as it shuffles the dimensions of the image and not sub-blocks/local block etc.

**Comment by Nic-Ma** [comment]
Hi @finalelement ,

Thanks for the feature request.
I am not very clear about the `shuffle` progress: do you mean to shuffle the pixels within selected block or shuffle the blocks in the entire image?

Thanks.

**Comment by finalelement** [comment]
Hi @Nic-Ma I mean to say the voxels/pixels are shuffled within a local patch. Please also share if you already see something similar already existing.

**Comment by Nic-Ma** [comment]
@finalelement , OK, we don't have such transform so far I think.
Please go ahead to implement your ideas.

Thanks in advance.

**Comment by Spenhouet** [comment]
There might also be a method similar to [RandomSwap](https://torchio.readthedocs.io/transforms/augmentation.html#randomswap) coming.
To differentiate I would propose the following naming:

- For the RandomSwap equivalent: `RandPatchSwap`
- For the patch shuffling proposed here: `RandPatchShuffle`

**Comment by Nic-Ma** [comment]
Hi @Spenhouet ,

Thanks for your explanation, naming sounds to me.
@finalelement Do you have any other comments?

Thanks.

**Comment by finalelement** [comment]
I've created a pull request. The code has been added in this [pull request](
https://github.com/Project-MONAI/MONAI/pull/2757).

This transform needs testcases and some more local testing, as well as how it has to be added to the dictionary.py

**Comment by finalelement** [comment]
[Follow this in future for making this N-d](https://github.com/Project-MONAI/MONAI/blob/6e025069bfc15ee325661601873f08f751695cda/monai/losses/deform.py#L34-L38)

Also shifting the transform from spatial to intensity, due to it being a better fit as an intensity transform.

**Comment by finalelement** [comment]
@wyli Thank you for all the support with this Issue. This current one supports 3D.

We will re-iterate on a new on a new PR with updated dev-branch for dictionary version and N-dimensional version of this transform.
